[16.0][MIG] fetchmail_attach_from_folder

NL66278 opened the pull request.

hbrunn committed:
[ADD] make retrieval configurable per folder; use msgid to exclude
duplicates; don't break existing configurations

hbrunn committed:
[MRG] lp:~therp-nl/therp-addons/fetchmail_attach_from_folder rev 89

NL66278 committed:
[IMP] - convert mailadresses to lowercase to increase change for match
      - handle situation that optional mail address componenents not present in mail
      - slight reorganisation of code to aid debugging and made pep8 compliant as well.

hbrunn committed:
[IMP] autopep8
[IMP] more efficient sorting of algorithms
[IMP] improve readability of help strings

hbrunn committed:
[ADD] msg_state field for folders - set the state of messages fetched
from imap servers
[IMP] clean up the folders' treeview

hbrunn committed:
[IMP] refactored fetch_mail
[IMP] also log when we're done with one folder

hbrunn committed:
[FIX] attach mail attachments to matched object in non-OpenERP case

hbrunn committed:
[IMP] make fetch_mail's refactoring more useful by returning the ids of
mails creates/objects matched
[ADD] preliminary docstrings

hbrunn committed:
[FIX] don't rollback() transation, use savepoint. This fixes concurrency
issues with long running cron threads as rollback also releases the lock

gurneyalex committed:
[REVERT] merged and pushed on 7.0 instead of 6.1 branch...

hbrunn committed:
[ADD] port fetchmail_attach_from_folder to 7.0

hbrunn committed:
[IMP] remove deprecated type field for views

hbrunn committed:
[IMP] adjust view definitions to 7.0
[FIX] use mail_thread.message_parse in wizard

sbidoul committed:
move all modules to __unported__ on master branch and set installable=False

hbrunn committed:
[REN] preliminary migration of fetchmail_attach_from_folder

hbrunn committed:
[IMP] make manual attach wizard usable again

hbrunn committed:
[ADD] placeholders

hbrunn committed:
[FIX] rename OpenERP->Odoo

hbrunn committed:
[IMP] readme

hbrunn committed:
[IMP] use a dedicated logger

hbrunn committed:
[FIX] we want object_id not to be required

hbrunn committed:
[ADD] test matching algorithms

yvaucher committed:
Add license key in __openerp__.py

gurneyalex committed:
Add OCA as author of OCA addons

In order to get visibility on https://www.odoo.com/apps the OCA board has
decided to add the OCA as author of all the addons maintained as part of the
association.

Commit:
Remove auto_install on fetchmail_attach_from_folder

The key 'active' is deprecated and is a synonym for 'auto_install'.
This addon should not be automatically installed.

yvaucher committed:
Add bug tracker link on README.rst

hbrunn committed:
[FIX] honor sequence

hbrunn committed:
[ADD] allow to deactivate folders

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
[UPD] prefix versions with 8.0

pedrobaeza committed:
[MIG] Make modules uninstallable

StefanRijnhart committed:
[FIX] Some fields are not there if the client asks for a search form

StefanRijnhart committed:
[RFR] Override hook

StefanRijnhart committed:
[IMP] Full email preview when attaching manually

gurneyalex committed:
[SEC] fetchmail_attach_from_folder: fix unsafe eval

pedrobaeza committed:
[MIG] Rename manifest files

NL66278 committed:
[MIG] fetchmail_attach_from_folder. Migration to 10.0.

NL66278 committed:
[FIX] fetchmail_attach_from_folder. Refactor for more testing.

NL66278 committed:
[FIX] fetchnail_attach_from_folder. Can not pass object to action context.

NL66278 committed:
[IMP] fetchmail_attach_from_folder. Separate confirm for folder.

Now it is impossible to confirm a connection to a imap server if there is a problem
in one folder. This commit provides a new state field on the folder and e separate
comfirm button per folder.

NL66278 committed:
[IMP] fetchmail_attach_from_folder. Show available folders.

Also do not reset connection status on change of underlying folders.

NL66278 committed:
[IMP] fetchmail_attach_from_folder. Button to retrieve single folder.

Also possibility to reset state for folder.

oca-travis committed:
[UPD] Update fetchmail_attach_from_folder.pot

ntsirintanis committed:
[FIX] replace simplejson dependency with json

OCA-git-bot committed:
fetchmail_attach_from_folder 10.0.1.1.0

mymage committed:
Translated using Weblate (Italian)

Currently translated at 2.5% (2 of 78 strings)

Translation: server-tools-10.0/server-tools-10.0-fetchmail_attach_from_folder
Translate-URL: https://translation.odoo-community.org/projects/server-tools-10-0/server-tools-10-0-fetchmail_attach_from_folder/it/

NL66278 committed:
[IMP] fetchmail_attach_from_folder: pre-commit stuff

NL66278 committed:
[MIG] fetchmail_attach_from_folder: Migration to 16.0

NL66278 committed:
[MIG] fetchmail_*: adapt tests

NL66278 committed:
[FIX] fetchmail_*: adapt to Odoo 16.0 and python 3.x

thomaspaulb commented:
@NL66278 Why the auto_backup fix? 16.0 is currently green also without that; so maybe it should be in a separate PR?

NL66278 committed:
[IMP] fetchmail_.._folder: optionally archive messages automatically

zamberjo commented:
Hi, better omit **auto_backup** changes, if you want do another PR

zamberjo commented on fetchmail_attach_from_folder/readme/CONTRIBUTORS.md:
```diff
@@ -0,0 +1,2 @@
+- Holger Brunn <hbrunn@therp.nl>
+- Ronald Portier <ronald@therp.nl>
```
This module was started by @gurneyalex in 7.0, I think you should continue to respect the authorship.

---

technical and functional review, just the subject I mentioned above

NL66278 replied:
The module was actually started by my former collegue @hbrunn , but as @gurneyalex  contributed some improvements I have no problem with adding him as a contributor.

anmarmo1 reviewed (approved):
Review functional is ok

NL66278 committed:
[IMP] fetch..._folder: Add contributor and maintainer

NL66278 commented:
@zamberjo I put auto_backup changes in own PR now: https://github.com/OCA/server-tools/pull/2939

NL66278 committed:
[FIX] fetchmaeil..folder: adapt test to current demo data

NL66278 committed:
[FIX] fetchmail..folder: handle non standard matching

Match algorithms other then 'odoo_standard' were not correctly dealt with in the migrated code. For
these algorithms we restore the handling (but with improvements) that was present in the 10.0 version.

gurneyalex commented:
/ocabot merge nobump
